http.js

```diff
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

Comment:
You make a compelling argument for guessing UTF-8 in the absence of a request override and absence of an explicit charset in the content type header. Guessing a charset should only be necessary for text/* MIME types, since application/* etc should provide Buffers.

I still argue that if the HTTP response content type calls for a charset we do not support, and if the requester does not provide a request.charset override, we would be doing a disservice by papering over correctness.

I do like explicitly checking whether the encoding is supported. That gives us an opportunity to provide a more informative error for HTTP read consumers.